documentation.md

```diff
@@ -0,0 +1,669 @@
+
+As read data come as a burst, and to allow the progressive release of the data in the burst, small counters are used to maintain the number of read data that can currently be released per internal identifier (_i.e._, per extended cell).
+The multi-hot release enable signal, similar to the one used for the write responses but with a possibly different width, has each bit set iff the corresponding counter is non-zero.
+
+When data is effectively released, the response banks notify this event using the _wrsp_released_iid_onehot_ and _rdata_released_iid_onehot_ one-hot signals.
+In the case of read data, the corresponding counter is decremented.
+In the case of write responses, the corresponding flip-flop is unset.
+
+<figure class="image">
+  <img src="https://i.imgur.com/zbh9VpO.png" alt="Delay calculator slots">
+  <figcaption>Fig: Delay calculator core: slots and release signals</figcaption>
+</figure>
+
+### Age management
+
+Age management is used in two point of the delay calculator core:
+
+- As many common scheduling strategies consider request relative age at some point, including the currently implemented strategy, the relative age between elementary burst entries must be (at least partially) maintained.
+- The oldest write slots must receive write data information first, as in AXI, write data requests correspond in order with the write address requests: the earlier write data requests correspond to the earlier write address requests.
+
+#### Write slot age matrix
+
+The write slot age matrix is a simple example of an age matrix.
+An element at index (i,j) is set iff the write slot j is older than i.
+Only the top-right part above the diagonal is stored.
+
+Each row is then masked with the _free_wslt_for_data_mhot_ multi-hot signal, which indicates whether each write slot is valid and able to host new write data entries, which is the case iff there is at least one unset bit in the corresponding _data_v_ signal.
+
+<figure class="image">
+  <img src="https://i.imgur.com/zRb7Ptp.png" alt="Age matrix">
+  <figcaption>Fig: Generic age matrix: the blue cells are stored in flip-flops</figcaption>
+</figure>
+
+#### Main age matrix
+
+The main age matrix side is the concatenation of two types of entries:
+
+- The _NumWSlots_ x _MaxBurstEffLen_$ elementary write burst entries, addressed as (slotId << log2(MaxBurstEffLen)) | eid, where eid is a notation, convenient here but not used in the source code.
+ - The _NumRSlots_ read data slots / read address requests.
+
+We have considered the fact that all read elementary burst entries in the same burst share the same age.
+
+### Finding optimal entries
+
+To find optimal the optimal entries to simulate new memory operations, the entries are first grouped by rank.
+All these groups, quotiented by rank, work in parallel without interaction.
```

Comment:
Maybe "All these per-rank groups work in parallel without interaction." ?

Reply:
Done!